core/modules/system/system.module

```diff
@@ -3003,6 +3003,11 @@ function system_header_block_form($settings) {
     '#empty_option' => '- ' . t('None') . ' -',
     '#default_value' => $settings['menu'],
   );
+  $form = array(
+    '#type' => 'fieldset',
+    '#title' => t('Site information components'),
+    '#description' => t('Here you can enable or disable the various components of the header of your site. To change the values of these components, use the !site_info page.', array('!site_info' => l(t('Site information'), 'admin/config/system/site-information'))),
+  );
```

Comment:
This needs to be moved above `$form['menu']`, otherwise it will remove the ability to select a different menu.